### DISCUSSION TITLE
NPM Package Converter

### DISCUSSION THREAD

**Discussion by ghost** [denoland/deno#8812] [Show and tell]
I have been reading up on the possibility of someone installing Deno to be used alongside NPM modules; but since that goes against what Deno is trying to stand for, I came up with the idea that there could be a converter using NPM's `package-lock.json` file (since it contains links for each dependency and is formatted conveniently in JSON). This is the basic function of how it would work, appending the links to the top of a Deno script. How it will be spread out upon request? That's a work in progress (though it could just target a `main/index.js` file.

Since people using NPM would use the program, it should be written in Node.js, of course.
```javascript
var prepend = require ('prepend-file'); //dependency
var fs = require('fs');

var file = JSON.parse(fs.readFile('package-lock.json'));

var dependenciesList = ''
for (var dependency of file.dependencies) {
  dependenciesList += 'import "' + dependency.resolved + '";\n';
}

var target = (process.argv[0] || 'index.js' || 'main.js');
prependFile.sync(target, dependenciesList);
```

**Comment by getspooky** [comment]
Hi F1nx
feel free to see this conversation https://github.com/denoland/deno/issues/47
